### PR TITLE
Get rid of `go-forceexport` dependency

### LINKED ISSUE #1: enabling `-race` flag for `go test` command panics with a fatal error
### Problem
As part of talon-one/talangv1#21 we would like to enable the `-race` flag as part of the CI tests step.
While doing so, we encountered the following output problem from this package. Running the `go test -count=1 -race .` command gives the same ouput:
```
fatal error: checkptr: unsafe pointer arithmetic

goroutine 1 [running, locked to thread]:
runtime.throw(0x14fa7fb, 0x23)
	/usr/local/opt/go@1.14/libexec/src/runtime/panic.go:1116 +0x72 fp=0xc000131d70 sp=0xc000131d40 pc=0x1073622
runtime.checkptrArithmetic(0xc000116f60, 0x0, 0x0, 0x0)
	/usr/local/opt/go@1.14/libexec/src/runtime/checkptr.go:43 +0xb5 fp=0xc000131da0 sp=0xc000131d70 pc=0x1045a95
github.com/talon-one/compare/vendor/github.com/alangpierce/go-forceexport.CreateFuncForCodePtr(0x146ee20, 0x17edc28, 0x11264a0)
	/Users/altjake/go/src/github.com/talon-one/compare/vendor/github.com/alangpierce/go-forceexport/forceexport.go:48 +0x1ed fp=0xc000131e90 sp=0xc000131da0 pc=0x11ff57d
github.com/talon-one/compare/vendor/github.com/alangpierce/go-forceexport.GetFunc(0x146ee20, 0x17edc28, 0x14f58e9, 0x16, 0x17a8dc0, 0xc000112260)
	/Users/altjake/go/src/github.com/talon-one/compare/vendor/github.com/alangpierce/go-forceexport/forceexport.go:20 +0x9b fp=0xc000131ed8 sp=0xc000131e90 pc=0x11ff36b
github.com/talon-one/compare.init.0()
	/Users/altjake/go/src/github.com/talon-one/compare/compare.go:29 +0x60 fp=0xc000131f28 sp=0xc000131ed8 pc=0x144c7a0
runtime.doInit(0x17aa800)
	/usr/local/opt/go@1.14/libexec/src/runtime/proc.go:5414 +0x8a fp=0xc000131f58 sp=0xc000131f28 pc=0x108287a
runtime.doInit(0x17a8bc0)
	/usr/local/opt/go@1.14/libexec/src/runtime/proc.go:5409 +0x57 fp=0xc000131f88 sp=0xc000131f58 pc=0x1082847
runtime.main()
	/usr/local/opt/go@1.14/libexec/src/runtime/proc.go:190 +0x1ce fp=0xc000131fe0 sp=0xc000131f88 pc=0x1075c5e
runtime.goexit()
	/usr/local/opt/go@1.14/libexec/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc000131fe8 sp=0xc000131fe0 pc=0x10a7101
FAIL	github.com/talon-one/compare	0.531s
FAIL
```

As Waleed mention in https://github.com/talon-one/talangv1/pull/21#issuecomment-703585239, it might be related to the use of `init` function for the package:
> I haven't encountered this issue before. I think you're right in saying it stems from the `compare` package. I don't think there's anything wrong with what it does though, since that call happens in `init` func which is only called once.

### PR DESCRIPTION
**Pull request by wlMalk**
## Problem
The package is using a dependency that is preventing us to upgrade to Go 1.16 and is not necessary.

## Solution
Modify the package so that `go-forceexport` is no longer needed by using `go:linkname`.

## Notes
- Should resolve #1 as well
- Switched to go mod

### CODE REVIEW

File: incomplete.s [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/*
+    This file is placed here to make sure the go compiler is running without the -complete option
+*/
```
Comment (mgfcmartinez) [inline incomplete.s]: missing terminal newline

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module github.com/talon-one/compare
+
+go 1.15
```
Comment (mgfcmartinez) [inline go.mod:3]: can we use 1.16 already?

**Review by mgfcmartinez — APPROVED** [review]
:tada:

**Review by Eun — APPROVED** [review]
Good idea by using linkname!

**Pull request merged**